### PR TITLE
Disabled Button SFX

### PR DESCRIPTION
**Pull request by gracebarsny**
**Overview**
1. Allowed UIButtons to play different SFX depending on whether they are interactable. This includes:
- AdoptButton on NavigationPanel
- Empty Dog Slots (not including Collar Slots)
- AdoptButton on DogShelterProfile
- GiftTimer when timer is active
- AdButton on MainMenu
- Filled Collar Slots
- ShopItemSlots when the item is unafforable
- CurrencyDisplays (to indicate they can't be interacted with)

2. Shortened the Daily Gift Timer in Tuning (change reflected in Drive)

3. Changed UISFXHandler to extend UIInteractable and implement IPointerDownHandler so click events aren't registered twice

### CODE REVIEW

File: Assets/Scripts/UI/PickupPup/DogSlot/DogSlot.cs [hunk 1]
```diff
@@ -91,8 +91,11 @@ protected override void setReferences()
 	{
 		base.setReferences();
 		button = ensureReference<UIButton>(searchChildren:true);
-		subscribeToUIButton();
-	}
+        images = GetComponentsInChildren<Image>();
+        dogImage = images[1];
```
Comment (imann24) [inline Assets/Scripts/UI/PickupPup/DogSlot/DogSlot.cs]: Can you fix this hardcoded var with a [SerializeField] ref instead?

File: Assets/Scripts/UI/PickupPup/DogSlot/DogSlot.cs [hunk 2]
```diff
@@ -216,11 +217,22 @@ void callOnFreeSlotClick()
 		}
 	}
 
+    protected void enable(bool isEnabled)
```
Comment (imann24) [inline Assets/Scripts/UI/PickupPup/DogSlot/DogSlot.cs]: This seems generic enough to justify moving to the superclass and overriding here

File: Assets/Scripts/UI/PickupPup/DogProfile/DogShelterProfile.cs [hunk 3]
```diff
@@ -24,6 +25,7 @@ protected override void setReferences()
     {
         base.setReferences();
         iconsObject.SetActive(false);
+        adoptUIButton = adoptButton.GetComponent<UIButton>();
```
Comment (imann24) [inline Assets/Scripts/UI/PickupPup/DogProfile/DogShelterProfile.cs]: This seems redundant. There should just be a single ref `UIButton adoptButton` and you should route any calls you need to the `Button` class through our custom `UIButton` class

File: Assets/Scripts/UI/Core/Button/UIButton.cs [hunk 4]
```diff
@@ -9,6 +9,11 @@
 
 public class UIButton : UIInteractable
 {
+    public bool Interactable
+    {
+        get { return button.interactable; }
```
Comment (imann24) [inline Assets/Scripts/UI/Core/Button/UIButton.cs]: Move brackets onto their own lines

**Comment by imann24** [comment]
@gracebarsny "Adopt" button is disabled on all screens. Is this intentional>

File: Assets/Scripts/UI/PickupPup/DogSlot/DogSlot.cs (hunk 1 above)
Comment (gracebarsny) [inline Assets/Scripts/UI/PickupPup/DogSlot/DogSlot.cs]: Funny thing is, this was already in the code. I just moved it to a different line. Will fix anyway.

File: Assets/Scripts/UI/PickupPup/DogProfile/DogShelterProfile.cs (hunk 3 above)
Comment (gracebarsny) [inline Assets/Scripts/UI/PickupPup/DogProfile/DogShelterProfile.cs]: UIButton does not provide any external functionality to adjust its image component. Its selectedColor and deselectedColor are hardcoded. Are you asking me to change that?

**Pull request merged**